ir/irutils.cpp

```diff
@@ -272,4 +273,36 @@ std::vector<const Expression *> flattenListOrStructExpression(const Expression *
                       listLikeExpr->node_type_name());
 }
 
+template <typename Stmts>
+const IR::Node *inlineBlockImpl(const Transform &t, Stmts &&stmts) {
+    if (stmts.size() == 1) {
+        // it could also be a declaration, and it that case, we need to wrap it in a block anyway
+        if (auto *stmt = (*stmts.begin())->template to<IR::Statement>()) {
+            return stmt;
+        }
+    }
+    if (t.getParent<const IR::Node>()->is<IR::BlockStatement>()) {
```

Comment:
could be just `if (t.getParent<IR::BlockStatement>())`

Reply:
Right, for some reason I thought `getParent` fails if the type is not right, I probably confused it with `getOriginal`.